### ISSUE TITLE
DiabetesProject

### ISSUE THREAD

**Issue by AyseSenaGencturk** [AyseSenaGencturk/MyPythonAIExercises#1]
# I took the database from the https://www.kaggle.com/datasets/uciml/pima-indians-diabetes-database?resource=download

"""
Created on Fri Sep  8 14:30:32 2023

@author: ayses
"""
# In[1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier

# Outcome = 1 diabete 
# Outcome = 0 healthy

data = pd.read_csv("diabetes.csv")
data.head()

# In[2]:

diatebespeople = data[data.Outcome == 1]
healthypeople = data[data.Outcome == 0]

# Now, we look only "gloucose" making a sample drawing.
plt.scatter(healthypeople.Age, healthypeople.Glucose, color="green", label="healthy", alpha = 0.4)
plt.scatter(diatebespeople.Age, diatebespeople.Glucose, color="red", label="diabete", alpha = 0.4)
plt.xlabel("Age")
plt.ylabel("Glucose")
plt.legend()
plt.show()

# In[3]:

# value y and x axis.
y = data.Outcome.values
x_puredata = data.drop(["Outcome"],axis=1)   
# We removed Outcome (dependent variable) and we took only independent variables.
# Since KNN algorihtm will be group in only x values.

# we did normalize.
x = (x_puredata - np.min(x_puredata ))/(np.max(x_puredata)-np.min(x_puredata ))

# before
print("Pure Datas before the normalization:\n")
print(x_puredata.head())

# after 
print("\n\n\n Datas of after the Normalization that we will give to AI:\n")
print(x.head())
    
# In[4]:

# We seperated with the train data and the test data.
# train data will be used for seperating the diabete and the healthy people
# test data will be used for "can the machine learning work true or not for seperating the diabete and healthy people.

x_train, x_test, y_train, y_test = train_test_split(x,y,test_size = 0.1,random_state=1)

# we set up the knn model.
knn = KNeighborsClassifier(n_neighbors = 3) # n_neighbors = k
knn.fit(x_train,y_train)
prediction = knn.predict(x_test)
print("for k=3 the results of the validation of test datas: ", knn.score(x_test, y_test))

# In[5]:

# what must k's value be?
# we find of the best value of k.
numerator = 1
for k in range(1,11):
    knn_new = KNeighborsClassifier(n_neighbors = k)
    knn_new.fit(x_train,y_train)
    print(numerator, "  ", "Accuracy rate: %", knn_new.score(x_test,y_test)*100)
    numerator += 1
    
# In[6]:

# Prediction about the new patient:
from sklearn.preprocessing import MinMaxScaler

# we normalized and we could be more speed for normalization, we used MinMax Scaler.
sc = MinMaxScaler()
sc.fit_transform(x_puredata)

new_prediction = knn.predict(sc.transform(np.array([[6,148,72,35,0,33.6,0.627,50]])))
new_prediction[0]
